codec-http2/src/main/java/io/netty/handler/codec/http2/Http2RemoteFlowController.java

```diff
@@ -53,6 +53,11 @@
 
         /**
          * Signal an error and release any retained buffers.
+         *<p>
+         * This method might be called multiple times.
```

Comment:
@nmittler correct. Moving the `complete()` call to when the promise completes is not feasible, because then multiple frames with END_STREAM could sneak in (as we have talked about last week).

Ideally, we should not have the `complete` method at all. It's more of a hack to get access to the `LifecycleManager` within the flow controller.